### PR TITLE
feature: create helm chart using the Azure KeyVault variant

### LINKED ISSUE #197: Create `tractusx-connector-azure-vault` chart
## WHAT

Create the `tractusx-connector-azure-vault` chart including the following tasks:

- create a copy of `tractusx-connector`
- update the Docker images 
- replace the Hashicorp Vault config with the Azure KeyVault config
- create chart tests
- add a matrix step in `verify.yaml` that executes the chart tests and some basic HTTP requests as outlined in the decision-record (#194)

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## WHAT

this PR creates a dedicated helm chart for the runtime variant that uses the Azure KeyVault

## WHY

Some stakeholders require AZKV and thus we need to provide and test it.

## FURTHER NOTES

This PR ~~will remain in `draft` until~~ can be merged once we either have a self-contained vault emulator, or an always-on vault instance

Closes #197 

[edit] successful run: https://github.com/paullatzelsperger/tractusx-edc/actions/runs/4869620950/jobs/8684354461

### CODE REVIEW

**Comment by paullatzelsperger** [comment]
FYI the deployment test will fail until we
- have an actual Az KeyVault to test
- provide credentials for it in the test

**Review by stefan-ettl — APPROVED** [review]

File: charts/tractusx-connector/README.md
Comment (florianrusch-zf) [inline charts/tractusx-connector/README.md:1]: Can you point out, why these changes are necessary? I guess someone forgot to regenerate the README in the past, right?

File: charts/tractusx-connector-azure-vault/templates/deployment-dataplane.yaml [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            timeoutSeconds: {{ .Values.dataplane.readinessProbe.timeoutSeconds }}
+            failureThreshold: {{ .Values.dataplane.readinessProbe.failureThreshold }}
+            successThreshold: {{ .Values.dataplane.readinessProbe.successThreshold }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.dataplane.resources | nindent 12 }}
+          env:
+          {{- if .Values.dataplane.debug.enabled }}
+            - name: "JAVA_TOOL_OPTIONS"
+            {{- if and .Values.dataplane.debug.enabled .Values.dataplane.debug.suspendOnStart }}
```
Comment (florianrusch-zf) [inline charts/tractusx-connector-azure-vault/templates/deployment-dataplane.yaml:101]:
```suggestion
            {{- if .Values.dataplane.debug.suspendOnStart }}
```

File: charts/tractusx-connector-azure-vault/templates/deployment-controlplane.yaml [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+            timeoutSeconds: {{ .Values.controlplane.readinessProbe.timeoutSeconds }}
+            failureThreshold: {{ .Values.controlplane.readinessProbe.failureThreshold }}
+            successThreshold: {{ .Values.controlplane.readinessProbe.successThreshold }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.controlplane.resources | nindent 12 }}
+          env:
+          {{- if .Values.controlplane.debug.enabled }}
+            - name: "JAVA_TOOL_OPTIONS"
+            {{- if and .Values.controlplane.debug.enabled .Values.controlplane.debug.suspendOnStart }}
```
Comment (florianrusch-zf) [inline charts/tractusx-connector-azure-vault/templates/deployment-controlplane.yaml:103]: Does this change (and the same for the dataplane) need to be applied to the other charts as well?

```suggestion
            {{- if .Values.controlplane.debug.suspendOnStart }}
```

File: charts/tractusx-connector-azure-vault/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+---
+apiVersion: v2
+name: tractusx-connector
```
Comment (florianrusch-zf) [inline charts/tractusx-connector-azure-vault/Chart.yaml:25]: If I understood it currently, everything else (java artefact, container image) will be published unter `...-azure-vault`, right? Why not this chart as well?

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

File: charts/tractusx-connector/README.md
Comment (paullatzelsperger) [reply charts/tractusx-connector/README.md:1]: TBH i don't recall, but that would be a strong guess :)

File: charts/tractusx-connector-azure-vault/Chart.yaml (hunk 3 above)
Comment (paullatzelsperger) [reply charts/tractusx-connector-azure-vault/Chart.yaml:25]: either an oversight, or a merge/rebase error

File: edc-tests/deployment/src/main/resources/helm/tractusx-connector-azure-vault-test.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    type: NodePort
+  endpoints:
+    management:
+      authKey: password
+  image:
+    pullPolicy: Never
+    tag: "latest"
+    repository: "edc-controlplane-postgresql-azure-vault"
+  securityContext:
+    # avoids some errors in the log: cannot write temp files of large multipart requests when R/O
```
[the remark this answers is not in the document]
Comment (paullatzelsperger) [reply edc-tests/deployment/src/main/resources/helm/tractusx-connector-azure-vault-test.yaml:53]: this is (part of) the stacktrace that results:
```
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault java.io.IOException: Read-only file system
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/java.io.UnixFileSystem.createFileExclusively(Native Method)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/java.io.File.createTempFile(Unknown Source)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at org.glassfish.jersey.media.multipart.internal.MultiPartReaderClientSide.createMimeConfig(MultiPartReaderClientSide.java:119)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at org.glassfish.jersey.media.multipart.internal.MultiPartReaderClientSide.<init>(MultiPartReaderClientSide.java:101)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at org.glassfish.jersey.media.multipart.internal.MultiPartReaderServerSide.<init>(MultiPartReaderServerSide.java:57)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at org.glassfish.hk2.utilities.reflection.ReflectionHelper.makeMe(ReflectionHelper.java:1356)
tx-prod-controlplane-f4f98f874-qcv27 tractusx-connector-azure-vault 	at org.jvnet.hk2.internal.ClazzCreator.createMe(ClazzCreator.java:272)
```
and looking at the `MultiPartReaderClientSide`, it seems to create a temp file to test whether the FS is readeable:
```java
//MultiPartReaderClientSide#L116ff
 // Validate - this checks whether it's possible to create temp files in currently set temp directory.
 try {
 //noinspection ResultOfMethodCallIgnored
     File.createTempFile("MIME", null, tempDir != null ? new File(tempDir) : null).delete();
 } catch (final IOException ioe) {
      LOGGER.log(Level.WARNING, LocalizationMessages.TEMP_FILE_CANNOT_BE_CREATED(properties.getBufferThreshold()), ioe);
 }
```

**Review by florianrusch-zf — APPROVED** [review]

File: edc-tests/deployment/src/main/resources/helm/tractusx-connector-azure-vault-test.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (florianrusch-zf) [reply edc-tests/deployment/src/main/resources/helm/tractusx-connector-azure-vault-test.yaml:53]: I'm with you! But I think a cleaner way would be to mount an `emptyDir` volume to the path where the `tempDir` points to. But we can do that in another Issue/PR. Could you create an Issue? I can then work on the PR for that.
